### DISCUSSION TITLE
PSFALCON list and add 2 tags

### DISCUSSION THREAD

**Discussion by lcagy** [CrowdStrike/psfalcon#350] [Q&A] [answered]
hi all

i new with psfalcon and have dficulties to create an PS script 

example: 

#Requires -Version 5.1
using module @{ModuleName='PSFalcon';ModuleVersion ='2.2'}
#Request-FalconToken -ClientId @@@@@@ -ClientSecret ##### -Cloud us-1

Find-FalconHostname -Path .\hostname.txt

Add-FalconGroupingTag -id -Tags FalconGroupingTags/tag1, FalconGroupingTags/tag2

Error when using "-id " 
```powershell
Line |
   8 |  Add-FalconGroupingTag -"id" -Tags FalconGroupingTags/tag1, FalconGrou …
     |                        ~~~~~
     | Cannot validate argument on parameter 'Id'. The argument "-id" does not match the "^[a-fA-F0-9]{32}$" pattern.
     | Supply an argument that matches "^[a-fA-F0-9]{32}$" and try the command again.
```

Error when using -id 
```powershell
Line |
   8 |  Add-FalconGroupingTag -id -Tags FalconGroupingTags/tag1, FalconGroupi …
     |                        ~~~
     | Missing an argument for parameter 'Id'. Specify a parameter of type 'System.String[]' and try again.
```

**Comment by bk-cs** [comment, marked as answer]
Unless GitHub is mangling your example, it looks like you are using `Find-FalconHostname` to find the required `device_id` values expected by the `Id` parameter of `Add-FalconGroupingTag`, but you're dropping them before you run the command.

Try one of these options:
```powershell
$HostList = Find-FalconHostname -Path .\hostname.txt
if ($HostList) { $HostList | Add-FalconGroupingTag -Tags FalconGroupingTags/tag1, FalconGroupingTags/tag2 }
```
```powershell
Find-FalconHostname -Path .\hostname.txt | Add-FalconGroupingTag -Tags FalconGroupingTags/tag1, FalconGroupingTags/tag2
```

**Comment by lcagy** [comment]
works with second option. 

tks a lot

**Comment by magasuan** [comment]
Hello!

In case you can help me.

How could the script be modified to read from a txt file all the IDs and apply the tag only to those hosts? 

thanks!

**Reply by magasuan** [reply]
I just realized that this script does exactly what I ask for? The only thing to note is the format of the txt. 

"Find-FalconHostname" can accept a list of hostnames as an array, or from a plain text file containing a hostname on each line. If a match is found, the hostname and device_id will be displayed, otherwise a warning message will be generated."

**Reply by bk-cs** [reply]
It expects a plaintext file with a hostname on each line. You can also use a different format and read it into an array which you pipeline to the command:
```powershell
Import-Csv .\devices.csv | Select-Object -ExpandProperty hostname | Find-FalconHostname
```
